src/test/integration/org/apache/cassandra/sidecar/common/JmxClientIntegrationTest.java

```diff
@@ -37,46 +40,77 @@ public class JmxClientIntegrationTest
+
+            testGossipInfo(jmxClient);
+
+            testCorrectVersion(jmxClient, String.valueOf(context.version.major));
+
+            testTableCleanup(jmxClient, context.cluster());
         }
     }
 
-    @CassandraIntegrationTest
-    void testGossipInfo(CassandraTestContext context) throws IOException
+    private void testGetOperationMode(JmxClient jmxClient, UpgradeableCluster cluster)
     {
-        try (JmxClient jmxClient = createJmxClient(context))
-        {
-            FailureDetector proxy = jmxClient.proxy(FailureDetector.class,
-                                                    "org.apache.cassandra.net:type=FailureDetector");
-            String rawGossipInfo = proxy.getAllEndpointStates();
-            assertThat(rawGossipInfo).isNotEmpty();
-            Map<String, ?> gossipInfoMap = GossipInfoParser.parse(rawGossipInfo);
-            assertThat(gossipInfoMap).isNotEmpty();
-            gossipInfoMap.forEach((key, value) -> GossipInfoParser.isGossipInfoHostHeader(key));
-        }
+        String opMode = jmxClient.proxy(SSProxy.class, SS_OBJ_NAME)
+                                 .getOperationMode();
+        assertThat(opMode).isNotNull();
+        assertThat(opMode).isIn("LEAVING", "JOINING", "NORMAL", "DECOMMISSIONED", "CLIENT");
+
+        IUpgradeableInstance instance = cluster.getFirstRunningInstance();
+        IInstanceConfig config = instance.config();
+        assertThat(jmxClient.host()).isEqualTo(config.broadcastAddress().getAddress().getHostAddress());
+        assertThat(jmxClient.port()).isEqualTo(config.jmxPort());
     }
 
-    @CassandraIntegrationTest
-    void testCorrectVersion(CassandraTestContext context) throws IOException
+    private void testGossipInfo(JmxClient jmxClient)
     {
-        try (JmxClient jmxClient = createJmxClient(context))
+        FailureDetector proxy = jmxClient.proxy(FailureDetector.class,
+                                                "org.apache.cassandra.net:type=FailureDetector");
+        String rawGossipInfo = proxy.getAllEndpointStates();
+        assertThat(rawGossipInfo).isNotEmpty();
+        Map<String, ?> gossipInfoMap = GossipInfoParser.parse(rawGossipInfo);
+        assertThat(gossipInfoMap).isNotEmpty();
+        gossipInfoMap.forEach((key, value) -> GossipInfoParser.isGossipInfoHostHeader(key));
+    }
+
+    private void testCorrectVersion(JmxClient jmxClient, String majorVersion)
+    {
+        String releaseVersion = jmxClient.proxy(SSProxy.class, SS_OBJ_NAME)
+                                         .getReleaseVersion();
+        assertThat(releaseVersion).startsWith(majorVersion);
+    }
+
+    // a test to ensure the jmx client can invoke the MBean method
+    private void testTableCleanup(JmxClient jmxClient, UpgradeableCluster cluster)
+    {
+        cluster.schemaChange("CREATE KEYSPACE jmx_client_test WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor' : 1}");
```

Comment:
can you really do a cleanup with a single node? although I'm not sure we want to test that the out of range token is executed, but rather than the cleanup interaction with JMX occurs